src/UnixSocket.php

```diff
@@ -0,0 +1,64 @@
+<?php
+namespace KS;
+
+class UnixSocket extends BaseSocket
+{
+    private $filename="";
+
+    public function __construct($socket, $filename)
+    {
+        parent::__construct($socket);
+        $this->filename = $filename;
+    }
+
+    public static function newUnixSocket($filename, $socketType=\SOCK_STREAM, $socketProtocol=0)
+    {
+        return new UnixSocket(parent::newRawSocket(\AF_UNIX, $socketType, $socketProtocol), $filename);
+    }
+
+    public function getFilename()
+    {
+        return $this->filename;
+    }
+
+    public function connect()
+    {
+        $result = \socket_connect($this->socket, $this->filename, 0);
+
+        return $result?Result::SUCCEEDED:Result::FAILED;

```

Comment:
Would be nice to have some spaces here.... Just a nit :). (i.e., `return $result ? Result::SUCCEEDED : Result::FAILED;`)